src/power_grid_model/core/power_grid_dataset.py

```diff
@@ -184,14 +184,11 @@ class CMutableDataset:
     _is_batch: bool
     _batch_size: int
     _mutable_dataset: MutableDatasetPtr
-    _buffer_views: List[CBuffer]
+    _buffer_views: list[CBuffer]
 
     def __new__(
         cls,
-        data: Union[
-            Mapping[ComponentType, np.ndarray],
-            Mapping[ComponentType, Union[np.ndarray, Mapping[str, np.ndarray]]],
-        ],
+        data: Mapping[ComponentType, np.ndarray] | Mapping[ComponentType, np.ndarray | Mapping[str, np.ndarray]],
```

Comment:
this one definitely can be replaced using a `TypeVar`. Maybe you can pull some inspiration from #663